proc_scripts/correlation_alignment.py

```diff
@@ -0,0 +1,149 @@
+from pylab import*
+from pyspecdata import *
+from scipy.optimize import leastsq, minimize, basinhopping
+import numpy as np
+from matplotlib.ticker import FuncFormatter
+logger= init_logging('info')
+@FuncFormatter
+def to_percent(y, position):
+    s = '%.2g'%(100 * y)
+    if rcParams['text.usetex'] is True:
+        return s + r'$\%$'
+    else:
+        return s + '%'
+def correl_align(s, align_phases=False,tol=1e-4,indirect_dim='indirect',fig_title='correlation alignment',ph1_selection=1,ph2_selection=1, sigma = 20,fl=None):
+    """
+    Align transients collected with chunked phase cycling dimensions along an indirect
+    dimension based on maximizing the correlation across all the transients and repeat
+    alignment until the calculated signal energy remains constant to within a given 
+    tolerance level.
+
+    Parameters
+    ==========
+    s:  nddata
+        an nddata object which contains phase cycle dimensions and an
+        indirect dimension
+    align_phases:   boolean
+    indirect_dim:   str
+                    name of the indirect dimension along which you seek to align
+                    the transients
+    fig_title:      str
+                    name for the figures generated
+    tol:            float
+                    sets the tolerance limit for the alignment procedure
+    ph1_selection:  int
+                    index position of the coherence pathway in phase program 1
+    ph2_selection:  int
+                    index position of the coherence pathway in phase program 2
+    sigma:          int
+                    sigma value for the gaussian fitting. Related to the linewidth
+                    of the given data.
```

Comment:
this should accept an fl argument (the figure list it uses to generate the figures) and a mask argument

Reply:
(the mask is for the masked norm)